src/AI/Muriel.java

```diff
@@ -960,15 +960,18 @@ private void queueUnitProductionActions(GameMap gameMap)
     // Build infantry from any remaining facilities.
     UnitModel infModel = myCo.cos[0].getUnitModel(UnitModel.TROOP);
     MapLocation loc = CPI.getLocationToBuild(infModel);
-    int infCost = loc.getOwner().getCost(infModel);
-    while( (budget >= infCost) && (CPI.availableUnitModels.contains(infModel)) )
+    int infCost = Integer.MAX_VALUE;
+    if( null != loc && null != loc.getOwner() )
+      infCost = loc.getOwner().getCost(infModel);
```

Comment:
Seems like the unit models should be owned by game scenario/rules/instance now, since this is some convoluted logic for something that should be very simple.